### ISSUE TITLE
Suchvorschläge: Nicht in alphab. Reihenfolge

### ISSUE THREAD

**Issue by wisydb** [wisydb/wisy#29]
Nur RLP?
Reproduzierbar?

**Comment by svenkaemper** [comment]
@wisydb Welche Suchvorschläge sind denn hier genau gemeint? Kursthemen? Anbieter? Kannst du eine Beispiel-URL nennen?

**Comment by svenkaemper** [comment]
@wisydb Siehe hierzu auch Issue #4 – sofern die Seite mit den Anbieter-Suchvorschlägen angegangen wird, sollte man auch die dort erwähnte "Vorfilterung" (nur "Träger" anzeigen) mit bedenken.

**Comment by debagel** [comment]
@wisydb Können wir das schließen? Wenn das Problem noch besteht bitte Beispiele auflisten.

**Comment by wisydb** [comment]
Nee. Hauptsächlich sind deutsche / französische / sonstige Anführungsstriche wohl die häufigsten Sonderzeichen, die die Reihenfolge beeinflussen, aber nicht nur.

Beispiel:
https://hamburg.kursportal.info/search?qtrigger=h&qs=Rosa+Luxemburg+Stiftung+Hamburg&qf=&filter_datum_von%5B%5D=&filter_preis_von%5B%5D=&filter_preis_bis%5B%5D=&filter_dauer_von%5B%5D=&filter_dauer_bis%5B%5D=&filter_tageszeit%5B%5D=&filter_foerderung%5B%5D=&filter_zielgruppe%5B%5D=&filter_metaabschlussart%5B%5D=&filter_unterrichtsart%5B%5D=&filter_tageszeit%5B%5D=&filter_volltext%5B%5D=&filter_order=t

DB:
titel: „Mein Bauch gehört mir“ | titel_sorted: „meinbauchgehoertmir“
titel: ‘Marea Verde‘. Feministische Bewegungen und (der Kampf um) die Legalisierung der Abtreibung in Argentinien | titel_sorted: ‘mareaverde‘feministischebewegungenundderkampfumdielegalisierungderabtreibunginargentinien

Stammt (wenn nicht aus dem Import) aus dem Sync-Renderer ruft auf Funktion "update_titel_sorted(...)" => ist in admin/config/trigger_kurse.inc.php definiert und => ruft auf Funktion "g_eql_normalize_natsort(..)" in admin/ecl.inc.php .

Dort werden ä wie a für Sortierung behandelt usw. das können wir ignorieren.

Es geht denke ich v.a. um die Ersetzung der (für die Sortierung) "überflüssigen" Zeichen, eben wie im Bespiel oben, also um die Zeilen:
// strip special characters
	$str = strtr($str,	'\'\\!°"§$%&/(){}[]=?+*~#,;.:-_<>|@©®£¥  ', '');

Wahrscheinlich könnte man nach jedem Auffalen, dass eine Sortierung "falsch" ist immer händisch hier ein Zeichen nachtragen.

Gibt es eine Möglichkeit entweder "alle bösen Zeichen" pauschal "wegzunehmen"? 
**Nachdem** also die Ersetzungen oben in der Funktion stattgefunden haben ("convert accented characters", "convert german umlaute", "convert numbers to a 'natural' sorting order") => müsste es nicht möglich sein zu sagen: und jetzt nur noch 0-9 oder A-Z oder a-z für die Sortierung?

**Comment by debagel** [comment]
@wisydb Ich habe jetzt in eql.inc.php/g_eql_normalize_natsort() am Ende alles außer "a-z", "A-Z", "0-9" rausgefiltert. Lokal sieht das Ergebnis gut aus. Ist es möglich das in der Sandbox zu testen?
